ambry-frontend/src/main/java/com.github.ambry.frontend/AmbrySecurityService.java

```diff
@@ -319,4 +329,33 @@ private void setAccountAndContainerHeaders(RestRequest restRequest, RestResponse
     }
     restResponseChannel.setHeader(RestUtils.Headers.PRIVATE, !container.isCacheable());
   }
+
+  /**
+   * Check precondtions for conditional delete if the {@code restRequest} contains the target
+   * account and container and they are not generic unknowns.
+   * @param restRequest the {@link RestRequest} that contains the {@link Account} and {@link Container} details.
+   * @throws RestServiceException if preconditions check failed.
+   */
+  private void checkForConditionalDelete(RestRequest restRequest) throws RestServiceException {
```

Comment:
can you put this away as a `public static` function in `RestUtils`? - call it something like `accountAndContainerNamePreconditionCheck()`